### PR TITLE
Added ckeditor plugin registry

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR adds the ability to register ckeditor plugins.

**Example:**

In a aura-extension:

```
sandbox.ckeditor.addPlugin(
    'internalLink',
    {
        init: function(editor) {
...
        }
    }
);
sandbox.ckeditor.addToolbarButton('links', 'InternalLink');
```

### CODE REVIEW

File: husky_extensions/ckeditor-extension.js [hunk 1]
```diff
@@ -23,7 +23,7 @@
         }
     });
 
-    define(['ckeditor', 'jqueryAdapter'], function() {
+    define(['underscore', 'ckeditor', 'jqueryAdapter'], function(_) {
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: Why do we need `'ckeditor'` and `'jqueryAdapter'` here, although it is not passed to the function?

File: husky_extensions/ckeditor-extension.js [hunk 2]
```diff
@@ -33,28 +33,155 @@
...
+        ToolbarBuilder.prototype.add = function(name, items) {
+            if (!this.toolbar[name]) {
+                this.toolbar[name] = [];
+            }
+
+            this.toolbar[name] = this.toolbar[name].concat(items);
+        };
+
+        /**
+         * Add toolbar items.
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: Wrong comment

File: husky_extensions/ckeditor-extension.js [hunk 3]
```diff
@@ -33,28 +33,155 @@
...
+         *
+         * @param {String} name
+         * @param {String[]} items
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: Is that `[]` specified in YUIDoc?

File: husky_extensions/ckeditor-extension.js [hunk 4]
```diff
@@ -33,28 +33,155 @@
...
+                this.toolbar[name] = [];
+            }
+
+            this.toolbar[name] = _.difference(this.toolbar[name], items);
+        };
+
+        /**
+         * Returns toolbar.
+         *
+         * @returns {{name, items}[]}
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: Is that kind of YUIDoc a real thing? Shouldn't it just be `Object`? You could define in the comment how the content looks in more detail.

File: husky_extensions/ckeditor-extension.js [hunk 5]
```diff
@@ -33,28 +33,155 @@
...
+
+                    addToolbarButton: function(toolbarName, button, icon) {
+                        if (!toolbar[toolbarName]) {
+                            toolbar[toolbarName] = [];
+                        }
+
+                        toolbar[toolbarName].push(button);
+
+                        if (!!icon) {
+                            icons.push(icon);
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: `icons` is an object, right? Then it probably doesn't have a `push` method.

File: husky_extensions/ckeditor-extension.js [hunk 6]
```diff
@@ -33,28 +33,155 @@
...
+                        }
+                    },
+
+                    getToolbar: function() {
+                        return JSON.parse(JSON.stringify(toolbar));
```
Comment (danrot) [inline husky_extensions/ckeditor-extension.js]: Use `this.sandbox.util.deepCopy` instead.

File: husky_extensions/ckeditor-extension.js (hunk 1 above)
Comment (wachterjohannes) [inline husky_extensions/ckeditor-extension.js]: we need it to be sure that it is loaded by require. this scripts will add some global vars and constructor's we need.

File: husky_extensions/ckeditor-extension.js (hunk 4 above)
Comment (wachterjohannes) [inline husky_extensions/ckeditor-extension.js]: we could use Object but then everything can be passed this is more specific. it says it is an object with the property name and items.

File: husky_extensions/ckeditor-extension.js (hunk 5 above)
Comment (wachterjohannes) [inline husky_extensions/ckeditor-extension.js]: thats right (:

File: husky_extensions/ckeditor-extension.js (hunk 4 above)
Comment (wachterjohannes) [inline husky_extensions/ckeditor-extension.js]: i have it from the google style guide https://google.github.io/styleguide/javascriptguide.xml#JsTypes

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**